test/core/origin_core/propose_block.js

```diff
@@ -0,0 +1,302 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: propose_block.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const web3 = require('../../test_lib/web3.js');
+
+const BN = require('bn.js');
+const EventsDecoder = require('../../test_lib/event_decoder.js');
+const Utils = require('../../test_lib/utils.js');
+
+const OriginCore = artifacts.require('OriginCore');
+
+contract('propose meta-block', async (accounts) => {
+
+
+    let originCore;
+    let auxiliaryCoreIdentifier = web3.utils.sha3("1");
+
+    let height, kernelHash, auxiliaryDynasty, auxiliaryBlockHash, gas,
+         originDynasty, originBlockHash, transactionRoot, stateRoot;
+
+    beforeEach(async () => {
+
+        height = 1,
+             kernelHash = web3.utils.sha3("1"),
+             auxiliaryDynasty = 50,
+             auxiliaryBlockHash = web3.utils.sha3("1"),
+             gas = 1000,
+             originDynasty = 1,
+             originBlockHash = web3.utils.sha3("1"),
+             transactionRoot = web3.utils.sha3("1");
+
+        let ost = accounts[0]
+             , initialValidators = [accounts[1], accounts[2]]
+             , validatorsWeights = [10, 20];
```

Comment:
I think we should agree to always put commas at the end of a line, never at the beginning. What do you think? @deepesh-kn ?